common/configuration/1password_provider.go

```diff
@@ -27,172 +25,156 @@ func (OnePasswordProvider) Role() Secrets                               { return
-	Value string `json:"value"`
+func (e notFoundError) Error() string {
+	return fmt.Sprintf("item %q not found in vault %q", e.ref, e.vault)
 }
 
-type fullResponse struct {
+// item is the JSON response from `op item get`.
+type item struct {
 	Fields []entry `json:"fields"`
 }
 
-// Decode a full item response from op
-func decodeFull(output []byte) ([]entry, error) {
-	var full fullResponse
-	if err := json.Unmarshal(output, &full); err != nil {
-		return nil, fmt.Errorf("error decoding op full response: %w", err)
-	}
-	return full.Fields, nil
+type entry struct {
+	ID    string `json:"id"`
+	Value string `json:"value"`
 }
 
-// Decode a single field from op
-func decodeSingle(output []byte) (*entry, error) {
-	var single entry
-	if err := json.Unmarshal(output, &single); err != nil {
-		return nil, fmt.Errorf("error decoding op single response: %w", err)
-	}
-	return &single, nil
+func (i item) value(field string) (string, bool) {
+	secret, ok := slices.Find(i.Fields, func(item entry) bool {
+		return item.ID == field
+	})
+	return secret.Value, ok
 }
 
-// Custom parser for 1Password secret references because the format is not a standard URL, and we also need to
-// allow users to omit the field name so that we can support secrets with multiple fields.
-//
-// Does not support "section-name".
-//
-//	op://<vault-name>/<item-name>[/<field-name>]
-//
-// Secret references are case-insensitive and support the following characters:
-//
-//	alphanumeric characters (a-z, A-Z, 0-9), -, _, . and the whitespace character
-//
-// If an item or field name includes a / or an unsupported character, use the item
-// or field's unique identifier (ID) instead of its name.
-//
-// See https://developer.1password.com/docs/cli/secrets-reference-syntax/
-type secretRef struct {
-	Vault string
-	Item  string
-	Field optional.Option[string]
-}
+// op --format json item get --vault Personal "With Spaces"
+func getItem(ctx context.Context, vault string, ref Ref) (*item, error) {
+	logger := log.FromContext(ctx)
 
-var validCharsRegex = regexp.MustCompile(`^[a-zA-Z0-9\-_\. ]+$`)
+	args := []string{"--format", "json", "item", "get", "--vault", vault, ref.String()}
+	output, err := exec.Capture(ctx, ".", "op", args...)
+	logger.Debugf("Getting item with args %v", args)
+	if err != nil {
+		if strings.Contains(string(output), "isn't a vault") {
+			return nil, fmt.Errorf("vault %q not found: %w", vault, err)
+		}
 
-func decodeSecretRef(ref string) (*secretRef, error) {
+		// Item not found, seen two ways of reporting this:
+		if strings.Contains(string(output), "not found in vault") {
+			return nil, notFoundError{vault, ref}
+		}
+		if strings.Contains(string(output), "isn't an item") {
+			return nil, notFoundError{vault, ref}
+		}
 
-	// Take out and check the "op://" prefix
-	const prefix = "op://"
-	if !strings.HasPrefix(ref, prefix) {
-		return nil, fmt.Errorf("must start with \"op://\"")
+		return nil, fmt.Errorf("run `op` with args %v: %w", args, err)
```

Comment:
I think this error is a bit confusing, and it also doesn't show the output, or is that why the error is as it is? Let's make it something more like `fmt.Errorf("%w: op %s\n%s", err, shellquote.Join(args...), output)`?

Also a general note - don't use `%v` for displaying slices to end users, it's [confusing](https://go.dev/play/p/nxGoWiKgYkh).

Reply:
I don't show the output because it might have contained the secret in the output. I will join the command for the log.